mealpy.py

```diff
@@ -1,126 +1,111 @@
-        return r.status_code
-
-    def get_cities(self):
-        if not self.cities:
-            r = requests.post(CITIES_URL, headers=self.headers)
-            self.cities = r.json()['result']
-        return self.cities
+        url = f'{BASE_URL}/1/login'
+        request = requests.post(url, data=json.dumps(data), headers=HEADERS)
+        self.cookies = request.cookies
+        self.cookies.set('isLoggedIn', 'true', domain=BASE_URL)
+        return request.status_code
+
+    @staticmethod
+    def get_cities():
+        url = f'{BASE_URL}/1/functions/getCitiesWithNeighborhoods'
+        request = requests.post(url, headers=HEADERS)
+        return request.json()['result']
 
     def get_city(self, city_name):
-        if not self.cities:
-            self.get_cities()
-        city = next((i for i in self.cities if i['name'] == city_name), None)
+        city = next((i for i in self.get_cities() if i['name'] == city_name), None)
         return city
 
-    def get_schedules(self, city_name, city_id=None):
-        if not city_id:
-            city_id = self.get_city(city_name)['objectId']
-        r = requests.get(
-            MENU_URL % city_id, headers=self.headers, cookies=self.cookies)
-        self.schedules = r.json()['schedules']
-        return self.schedules
-
-    def get_schedule_by_restaurant_name(
-            self, restaurant_name, city_name=None, city_id=None):
-        if not self.schedules:
-            self.get_schedules(city_name, city_id)
-        restaurant = next(i for i in self.schedules if i['restaurant']['name'] == restaurant_name)
+    def get_schedules(self, city_name):
+        city_id = self.get_city(city_name)['objectId']
+        url = f'{BASE_URL}/api/v1/cities/%s/product_offerings/lunch/menu'
+        request = requests.get(url % city_id, headers=HEADERS, cookies=self.cookies)
+        return request.json()['schedules']
+
+    def get_schedule_by_restaurant_name(self, restaurant_name, city_name):
+        restaurant = next(
+            i
+            for i in self.get_schedules(city_name)
+            if i['restaurant']['name'] == restaurant_name
+        )
         return restaurant
 
-    def get_schedule_by_meal_name(
-            self, meal_name, city_name=None, city_id=None):
-        if not self.schedules:
-            self.get_schedules(city_name, city_id)
-
-        return next(i for i in self.schedules if i['meal']['name'] == meal_name)
+    def get_schedule_by_meal_name(self, meal_name, city_name):
+        return next(i for i in self.get_schedules(city_name) if i['meal']['name'] == meal_name)
 
     def reserve_meal(
-            self, timing, restaurant_name=None, meal_name=None, city_name=None,
-            city_id=None, cancel_current_meal=False):
+            self,
+            timing,
+            city_name,
+            restaurant_name=None,
+            meal_name=None,
+            cancel_current_meal=False,
+    ):  # pylint: disable=too-many-arguments
```

Comment:
Don't worry about this override: #4 will introduce argparse, which captures all these more effectively (the mutually exclusive `restaurant_name` and `meal_name` is a code smell that argparse can solve).

Reply:
I should also add context that pylint defaults are aggressive (imo) and hit false positives. i.e. there are often cases where adding abstraction to keep under the pylint limit of 5 params is "foolish consistency".